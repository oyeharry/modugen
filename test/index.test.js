const path = require('path');
const pify = require('pify');
const fs = require('fs');
const del = require('del');
const modugen = require('..');

const fsAsync = pify(fs);
const destPath = './template-test';

beforeAll(async () => {
  await del(['./template-test']);
});

test('Verify module generation without moduleName directory', async () => {
  await modugen('src.CoreModule.Button', {
    destPath,
    templateFilesDir: './test/templates',
  });

  const moduleFile = path.resolve(path.join(destPath, 'src/Button.jsx'));
  const moduleTestFile = path.resolve(
    path.join(destPath, 'src/Button.test.js')
  );
  const moduleIndexFile = path.resolve(path.join(destPath, 'src/index.js'));

  const moduleFileContent = await fsAsync.readFile(moduleFile, 'utf-8');
  const moduleTestFileContent = await fsAsync.readFile(moduleTestFile, 'utf-8');
  const moduleIndexFileContent = await fsAsync.readFile(
    moduleIndexFile,
    'utf-8'
  );

  expect(moduleFileContent).toBeTruthy();
  expect(moduleTestFileContent).toBeTruthy();
  expect(moduleIndexFileContent).toBeTruthy();

  expect(moduleFileContent).toMatchSnapshot();
  expect(moduleTestFileContent).toMatchSnapshot();
  expect(moduleIndexFileContent).toMatchSnapshot();
});

test('Verify module generation with moduleName file and directory', async () => {
  await modugen('CoreModule.BaseButton', {
    destPath,
    templateFilesDir: './test/templates',
  });

  const moduleWithoutDirFile = path.resolve(
    path.join(destPath, '/BaseButton.js')
  );

  const moduleFile = path.resolve(
    path.join(destPath, 'base-button/BaseButton.jsx')
  );
  const moduleTestFile = path.resolve(
    path.join(destPath, 'base-button/test/BaseButton.test.js')
  );
  const moduleIndexFile = path.resolve(
    path.join(destPath, 'base-button/index.js')
  );

  const moduleReadMeFile = path.resolve(path.join(destPath, '/README.md'));

  const moduleTodosFile = path.resolve(path.join(destPath, '/Todos.txt'));

  const moduleWithoutDirFileContent = await fsAsync.readFile(
    moduleWithoutDirFile,
    'utf-8'
  );
  const moduleFileContent = await fsAsync.readFile(moduleFile, 'utf-8');
  const moduleTestFileContent = await fsAsync.readFile(moduleTestFile, 'utf-8');
  const moduleIndexFileContent = await fsAsync.readFile(
    moduleIndexFile,
    'utf-8'
  );
  const moduleReadMeFileContent = await fsAsync.readFile(
    moduleReadMeFile,
    'utf-8'
  );
  const moduleTodosFileContent = await fsAsync.readFile(
    moduleTodosFile,
    'utf-8'
  );

  expect(moduleWithoutDirFileContent).toBeTruthy();
  expect(moduleFileContent).toBeTruthy();
  expect(moduleTestFileContent).toBeTruthy();
  expect(moduleIndexFileContent).toBeTruthy();
  expect(moduleReadMeFileContent).toBeTruthy();
  expect(moduleTodosFileContent).toBeTruthy();

  expect(moduleWithoutDirFileContent).toMatchSnapshot();
  expect(moduleFileContent).toMatchSnapshot();
  expect(moduleTestFileContent).toMatchSnapshot();
  expect(moduleIndexFileContent).toMatchSnapshot();
  expect(moduleReadMeFileContent).toMatchSnapshot();
  expect(moduleTodosFileContent).toMatchSnapshot();
});
