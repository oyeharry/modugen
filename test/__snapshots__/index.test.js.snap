// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify module generation with moduleName file and directory 1`] = `
"// @flow
import React from 'react';
import PropTypes from 'prop-types';

type Props = {
  label?: string,
  onClick?: Function,
};

function BaseButton(props) {
  return <button>My Button</button>;
}

export default BaseButton;
"
`;

exports[`Verify module generation with moduleName file and directory 2`] = `
"import React from 'react';
import PropTypes from 'prop-types';

const baseButton = props => {
  return <button>My Button</button>;
};

export default baseButton;
"
`;

exports[`Verify module generation with moduleName file and directory 3`] = `
"import BaseButton from 'BaseButton';

test('verify BaseButton component', () => {});
"
`;

exports[`Verify module generation with moduleName file and directory 4`] = `
"export { default } from './BaseButton';
"
`;

exports[`Verify module generation with moduleName file and directory 5`] = `
"## Following are case formats

BaseButton
BaseButton
baseButton
base-button
base-button
base-button
base_button
base.button
BASE_BUTTON
base button
Base Button
"
`;

exports[`Verify module generation with moduleName file and directory 6`] = `
"#1 This is BaseButton Todo 1
#2 This is BaseButton Todo 2
"
`;

exports[`Verify module generation without moduleName directory 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';

function Button(props) {
  return <button>My Button</button>;
}

export default Button;
"
`;

exports[`Verify module generation without moduleName directory 2`] = `
"import Button from 'Button';

test('verify Button component', () => {});
"
`;

exports[`Verify module generation without moduleName directory 3`] = `
"export { default } from './Button';
"
`;
